server/src/main/java/org/elasticsearch/search/aggregations/Aggregator.java

```diff
@@ -91,6 +93,53 @@ public static boolean descendsFromBucketAggregator(Aggregator parent) {
      */
     public abstract Aggregator subAggregator(String name);
 
+    /**
+     * Resolve the next step of the sort path as though this aggregation
+     * supported sorting. This is usually the "first step" when resolving
+     * a sort path because most aggs that support sorting their buckets
+     * aren't valid in the middle of a sort path.
+     * <p>
+     * For example, the {@code terms} aggs supports sorting its buckets, but
+     * that sort path itself can't contain a different {@code terms}
+     * aggregation.
+     */
+    public final Aggregator resolveSortPathOnValidAgg(AggregationPath.PathElement next, Iterator<AggregationPath.PathElement> path) {
```

Comment:
I'm not super comfortable with this name, but I think it'll fall away in a followup change. When I dig into that last `instanceof` in `resolveTopmostAggregator`.